src/WebAPI/Controllers/RestaurantController.cs

```diff
@@ -16,15 +16,20 @@ public RestaurantController(IRecommendationRepository recommendationRepository)
             _recommendationRepository = recommendationRepository;
         }
 
+
         [HttpGet]
-        public Task<IEnumerable<Restaurant>> Get()
+        public Task<IEnumerable<Restaurant>> Get(int pageOffset, int pageLimit)
         {
-            return GetRestaurants();
+            return GetRestaurants(pageOffset, pageLimit);
         }
 
-        private async Task<IEnumerable<Restaurant>> GetRestaurants()
+        private async Task<IEnumerable<Restaurant>> GetRestaurants(int pageOffset, int pageLimit)
         {
-            return await _recommendationRepository.GetAllRestaurants();
+            if (pageOffset != 0 || pageLimit != 0) {
```

Comment:
This should probably be pageoffset > 0 && pagelimit > 0

Reply:
Would you ever use them individually?

For example,

I only want to see ten items, no paging required

Reply:
It was more for the negative values that could be passed in, you could technically do pageOffset > 0 || pageLimit > 0 but I don't see the value in it because these params are tightly coupled, unless I'm misunderstanding you.